01_fib/fibonacci.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func calculateFib(n int) []int {
+	var (
+		fn   = 1
+		f0   int
+		f1   = 1
+		list []int
+	)
+
+	for i := 0; i < n; i++ {
+		if i > 0 {
+			fn = f0 + f1
+			f0, f1 = f1, fn
+		}
+		list = append(list, fn)
+	}
+	return list
```

Comment:
Love the slice as return value. However, this could still be simplified:
```go
	a, b, result := 1, 1, []int{}
	for i := 1; i <= n; i++ {
		result = append(result, a)
		a, b = b, a+b
	}
	return result
```

Reply:
thank you, this looks neat, updated.